File: src/3194.minimum-average-of-smallest-and-largest-elements.3471/solution.ts
function minimumAverage (nums: number[]): number {
    nums.sort((a, b) => a - b);
    const averages:number[] = [];
    let left = 0;
    let right = nums.length - 1;
    while (left < right) {
        averages.push((nums[left++] + nums[right--]) / 2);
    }
    return Math.min(...averages);
}

File: src/3194.minimum-average-of-smallest-and-largest-elements.3471/solution.java
class Solution {
    public double minimumAverage(int[] nums) {
        double result = Double.MAX_VALUE;
        Arrays.sort(nums);
        int left = 0;
        int right = nums.length-1;
        while (left<right){
            result = Math.min(result, (nums[left++]+nums[right--])/2.0  );
        }
        return result;
    }
}